File: backend/internal/model/team.go
package model

import "time"

// Team 参赛团队实体。
type Team struct {
	ID        int64     `gorm:"primaryKey" json:"id"`
	Name      string    `gorm:"size:80;uniqueIndex;not null" json:"name"`
	Slogan    string    `gorm:"size:255" json:"slogan"`
	CaptainID int64     `gorm:"index;not null" json:"captain_id"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`

	Members []TeamMember `gorm:"foreignKey:TeamID" json:"members,omitempty"`
}

// TeamMember 团队成员关联实体。
type TeamMember struct {
	ID        int64     `gorm:"primaryKey" json:"id"`
	TeamID    int64     `gorm:"index;not null;uniqueIndex:idx_team_user" json:"team_id"`
	UserID    int64     `gorm:"index;not null;uniqueIndex:idx_team_user" json:"user_id"`
	Role      string    `gorm:"size:20;default:member;not null" json:"role"`
	JoinedAt  time.Time `json:"joined_at"`
}

// Registration 团队报名活动实体。
type Registration struct {
	ID           int64      `gorm:"primaryKey" json:"id"`
	TeamID       int64      `gorm:"index;not null;uniqueIndex:idx_team_activity" json:"team_id"`
	ActivityID   int64      `gorm:"index;not null;uniqueIndex:idx_team_activity" json:"activity_id"`
	Status       string     `gorm:"size:20;default:pending;not null;index" json:"status"`
	StartTime    *time.Time `json:"start_time"`
	FinishTime   *time.Time `json:"finish_time"`
	TotalSeconds int        `gorm:"default:0;not null" json:"total_seconds"`
	RegisteredAt time.Time  `json:"registered_at"`
	UpdatedAt    time.Time  `json:"updated_at"`
}

File: frontend/src/api/team.ts
import request from '../utils/request'

export interface TeamMember {
  user_id: number
  username: string
  nickname: string
  role: string
}

export interface Team {
  id: number
  name: string
  slogan?: string
  captain_id: number
  member_count: number
  members?: TeamMember[]
  created_at: string
}

export interface Registration {
  id: number
  team_id: number
  team_name: string
  activity_id: number
  activity_title: string
  status: string
  start_time?: string
  finish_time?: string
  total_seconds: number
  duration: string
  registered_at: string
}

export function createTeam(data: { name: string; slogan?: string }) {
  return request.post('/teams', data)
}

export function listMyTeams() {
  return request.get<Team[]>('/teams/mine')
}

export function listTeams(params: { page?: number; page_size?: number }) {
  return request.get<{ list: Team[]; total: number }>('/teams', { params })
}

export function getTeam(id: number) {
  return request.get<Team>(`/teams/${id}`)
}

export function joinTeam(teamId: number) {
  return request.post(`/teams/${teamId}/join`, { team_id: teamId })
}

export function leaveTeam(teamId: number) {
  return request.delete(`/teams/${teamId}/leave`)
}

export function applyActivity(data: { team_id: number; activity_id: number }) {
  return request.post('/registrations', data)
}

export function listMyRegistrations() {
  return request.get<Registration[]>('/registrations/mine')
}

export function listRegistrationsByActivity(activityId: number) {
  return request.get<Registration[]>(`/activities/${activityId}/registrations`)
}

export function approveRegistration(id: number) {
  return request.post(`/registrations/${id}/approve`)
}

export function rejectRegistration(id: number) {
  return request.post(`/registrations/${id}/reject`)
}
